docs/src/main/asciidoc/hibernate-search-guide.adoc

```diff
@@ -0,0 +1,627 @@
+All the upcoming updates coming through Hibernate ORM operations will be synchronized automatically to the full text index.
+If you don't import data manually in the database, you don't need that:
+the mass indexer should then only be used when you change your indexing configuration (adding a new field, changing an analyzer's configuration...) and you want the new configuration to be applied to your existing entities.
+<3> This is where the magic begins: just adding the annotations to our entities make them available for full text search: we can now query the index using the Hibernate Search DSL.
+<4> First, we get an Hibernate Search session from the injected entity manager.
+<5> We indicate that we are searching for ``Author``s.
+<6> We create a predicate: if the pattern is empty, we use a `matchAll()` predicate.
+<7> If we have a valid pattern, we create a https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html[`simpleQueryString()`] predicate on the `firstName`, `lastName` and `books.title` fields matching our pattern.
+<8> We define the sort order of our results. Here we sort by last name, then by first name. Note that we use the specific fields we created for sorting.
+<9> This is self explanatory. Obviously, paging is also supported.
+
+[NOTE]
+====
+Hibernate Search supports a significant part of the Elasticsearch predicates (match, range, nested, phrase, spatial...).
+Feel free to explore the DSL by using autocompletion.
+====
+
+== Configuring the application
+
+As usual, we can configure everything in the Quarkus configuration file, `application.properties`.
+
+Edit `src/main/resources/application.properties` and inject the following configuration:
+
+[source,properties]
+----
+quarkus.datasource.url=jdbc:postgresql:quarkus_test # <1>
+quarkus.datasource.driver=org.postgresql.Driver
+quarkus.datasource.username=quarkus_test
+quarkus.datasource.password=quarkus_test
+
+quarkus.hibernate-orm.database.generation=drop-and-create # <2>
+
+quarkus.hibernate-search.elasticsearch.version=7 # <3>
+quarkus.hibernate-search.elasticsearch.analysis-configurer=org.acme.hibernate.search.elasticsearch.config.AnalysisConfigurer # <4>
+quarkus.hibernate-search.elasticsearch.automatic-indexing.synchronization-strategy=searchable # <5>
+quarkus.hibernate-search.elasticsearch.index-defaults.lifecycle.strategy=drop-and-create # <6>
+quarkus.hibernate-search.elasticsearch.index-defaults.lifecycle.required-status=yellow # <7>
+----
+<1> Let's create a PostgreSQL datasource.
+<2> We will drop and recreate the schema every time we start the application.
+<3> We need to inform Hibernate Search from the version of Elasticsearch we will use.
+It does not need to be very precise but it is important as the Elasticsearch mapping is created at build time and thus is not able to connect to the cluster to automatically detect the version.
```

Comment:
Maybe rather "it is important because there are significant differences between Elasticsearch mapping syntax depending on the version. Since the mapping is created at build time to reduce startup time, Hibernate Search cannot connect to the cluster to automatically detect the version"

Reply:
Done.